fastlane/lib/util.rb

```diff
@@ -45,3 +45,76 @@ def travis?
 def circle?
   ENV['CIRCLECI'] == 'true'
 end
+
+# diff two hashes at a path and return the changed items
+def deps_diff(old_hash, new_hash, path)
+  (new_hash.dig(*path).to_a - old_hash.dig(*path).to_a).to_h
+end
+
+# checks if the native build needs to be run
+def should_build?
```

Comment:
RuboCop wants this to get split up a bit; it's a bit long, and the ABC and Cyclomatic complexity are a bit high. But this is a very good start.

Reply:
I pulled the big block in the middle out into a separate function; hopefully that helps